feat(KIE): Override KIE KogitoEditorEnvelope

lordrip opened the pull request:
### Context
Currently, the `kie-tools` repository uses `React 17` which has a particular way to be bootstrapped, referenced now as `Legacy Root API`.

Kaoto UI uses `React 18` which uses a different bootstrap API called `New Root API`, and while it's backward compatible, the runtime behavior is the one from `React 17`.

This causes many subtles differences between how Kaoto behaves as a web deployment vs how it behaves as a VS Code target.

Updating `React` in the [`kie-tools`](https://github.com/apache/incubator-kie-tools) it's not viable at the moment, see: https://github.com/apache/incubator-kie-issues/issues/118

### Changes
This commit copies the `KogitoEditorEnvelope.tsx` and the `index.ts` file into this repository so we can override how React is bootstrapped to use the `New Root API`.

### Notes
* https://react.dev/blog/2022/03/08/react-18-upgrade-guide#updates-to-client-rendering-apis
* https://github.com/reactwg/react-18/discussions/5

cc @fantonangeli

fix: https://github.com/KaotoIO/vscode-kaoto/issues/1074

lordrip committed:
feat(KIE): Override KIE KogitoEditorEnvelope

Currently, the `kie-tools` repository uses `React 17` which has a
particular way to be bootstrapped, referenced now as `Legacy Root API`.

Kaoto UI uses `React 18` which uses a different bootstrap API called
`New Root API`, and while it's backward compatible, the runtime behavior
is the one from `React 17`.

This causes many subtles differences between how Kaoto behaves as a web
deployment vs how it behaves as a VS Code target.

Updating `React` in the [`kie-tools`](https://github.com/apache/incubator-kie-tools) it's not viable at the moment, see: https://github.com/apache/incubator-kie-issues/issues/118

This commit copies the `KogitoEditorEnvelope.tsx` and the `index.ts`
file into this repository so we can override how React is bootstrapped
to use the `New Root API`.

Notes:
* https://react.dev/blog/2022/03/08/react-18-upgrade-guide#updates-to-client-rendering-apis
* https://github.com/reactwg/react-18/discussions/5

relates: https://github.com/KaotoIO/vscode-kaoto/issues/1074

apupier reviewed (changes requested):
tests related to custom catalogs are failing.

EDIT: Testing manually, at first sight the catalog are loading correctly. so might be just a UI test to adjust

djelinek commented:
in case you will need some help with failing UI tests, let me know 🙂

lordrip committed:
chore(deps): Update @patternfly/react-core dependency

Introduces the fix for the issue by updating `@patternfly/*` dependencies

fix: KaotoIO/kaoto#2501
fix: KaotoIO/kaoto#2535
relates: patternfly/patternfly-react#11993
relates: https://github.com/KaotoIO/vscode-kaoto/issues/1074
